### PR TITLE
feat: Http client settings for xxx-proxy adapters

### PR DESCRIPTION
**Pull request by dgarus**
Close: #1375 

- ability to define `http` client setting in `meta` and `repo` configs
- ability to define several proxy servers 
- using of `https.proxyHost` property

### CODE REVIEW

File: artipie-main/src/main/java/com/artipie/SliceFromConfig.java [hunk 1]
```diff
@@ -307,4 +310,10 @@ private static Slice build(
             )
             .get();
     }
+
+    private static ClientSlices clientSlices(final RepoConfig cfg, final Settings settings) {
+        return new JettyClientSlices(
```
[the remark this answers is not in the document]
Comment (olenagerasimova) [reply artipie-main/src/main/java/com/artipie/SliceFromConfig.java:315]: Also, should not `JettyClientSlices` it be started before usage?

**Review by olenagerasimova — COMMENTED** [review]
@dgarus please, take a look: some typos and important question

File: artipie-main/src/main/java/com/artipie/SliceFromConfig.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dgarus) [reply artipie-main/src/main/java/com/artipie/SliceFromConfig.java]: @olenagerasimova thanks, a good point.

**Comment by dgarus** [comment]
@olenagerasimova 
I added `SlicesCache` that manages `JettyClientSlices`.
Please take a look.

File: artipie-main/src/main/java/com/artipie/RepositorySlices.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+     * @return Slice for repo
+     */
+    private SliceValue resolve(final Key name, final int port) {
+        final RepoConfig cfg = new RepositoriesFromStorage(this.settings)
+            .config(name.string()).toCompletableFuture().join();
+        if (cfg.port().isEmpty() || cfg.port().getAsInt() == port) {
+            final JettyClientSlices client = new JettyClientSlices(
+                cfg.httpClientSettings().orElseGet(settings::httpClientSettings)
+            );
+            client.start();
```
Comment (olenagerasimova) [inline artipie-main/src/main/java/com/artipie/RepositorySlices.java:97]: hm, http client is required for proxy repository only, so why create it for each repository?

**Review by olenagerasimova — COMMENTED** [review]
great, although I have one question

File: artipie-main/src/main/java/com/artipie/RepositorySlices.java (hunk 2 above)
Comment (dgarus) [reply artipie-main/src/main/java/com/artipie/RepositorySlices.java:97]: A good point!
I'm going to implement a more suitable approach in #1386

Comment (olenagerasimova) [reply artipie-main/src/main/java/com/artipie/RepositorySlices.java:97]: for now, it's necessary to make sure all the integration tests in artipie pass... There are tests with proxies, so we'll know for sure if smth is broken

**Review by olenagerasimova — APPROVED** [review]

**Pull request merged**